test/e2e/cloudstack_test.go

```diff
@@ -0,0 +1,35 @@
+//go:build e2e
+// +build e2e
+
```

Comment:
I'm not a fan of this file name. It's too generic. Why can't we put this test in the labels test file like it was before? That seemed fitting. We could just add a comment indicating it's a specific feature for the CloudStack provider

Reply:
I changed back and forth in my local for the location of that piece of code. 

1. node name matching capi machine name is a requirements for cloudstack, other providers may not follow this rule (vsphere did follow, not sure other providers). Since it is provider specific, I choose to have a cloudstack file name, just like tinkerbell_test.go
2. label_test.go is code related to labels, so I believe the node name matching validation logic does not belong there.

failuredomain node label is label related and cloudstack specific, so I am open to put it inside cloudstack_test.go or label_test.go

Instead of cteating 2 e2e to test node label and node name separately, I combined them together into cloudstack_test.go: 
```
test.ValidateControlPlaneNodes(framework.ValidateControlPlaneFailureDomainLabels, framework.ValidateControlPlaneNodeNameMatchCAPIMachineName)
	test.ValidateWorkerNodes(framework.ValidateWorkerNodeFailureDomainLabels, framework.ValidateWorkerNodeNameMatchCAPIMachineName)
	```

Reply:
Even though this is CloudStack specific, a lot of tests are cloudstack specific. These files are intentionally organized by feature rather than provider. Since we include the provider name in the test, that itself should be sufficiently distinguishable, together with a comment, to indicate that this feature is Cloudstack-specific.

Reply:
Alright, I'll defer to @g-gaston on this one - I'm fine either way.

Reply:
what about `cloudstack_taints_test.go`?
If the "flow" for cloudstack and other providers for this tested scenario is not the same, I'm no against having a different file. However, `cloudstack_test` seems very broad and risks becoming a huge dumping ground for all cloudstack tests. Which makes reading the tests quite difficult.

Reply:
It is good. I can change it to cloudstack_labels_test.go - we are not testing cloudstack taint here.